File: src/main/java/ru/darkkeks/vkmirror/autoreg/SetDescriptionAction.java
package ru.darkkeks.vkmirror.autoreg;

public class SetDescriptionAction extends BotFatherAction<Void> {

    private static final String SET_DESCRIPTION = "/setdescription";

    private String username;
    private String description;

    public SetDescriptionAction(BotAutoReg autoReg, String username, String description) {
        super(autoReg);

        this.username = username;
        this.description = description;
    }

    @Override
    public void execute() {
        String newBot = getAutoReg().executeCommand(SET_DESCRIPTION);

        int timeout = getAutoReg().getTimeoutSeconds(newBot);
        if (timeout != BotAutoReg.NO_TIMEOUT) {
            getCompletableFuture().completeExceptionally(new BotFatherTimeoutException(timeout));
            return;
        }

        String afterUsername = getAutoReg().executeCommand(username);

        if(BotAutoReg.INVALID_BOT_SELECTED.matcher(afterUsername).find()) {
            getCompletableFuture().completeExceptionally(new IllegalStateException("Invalid username"));
        }

        String result = getAutoReg().executeCommand(description);

        finish(null);
    }
}

File: src/main/java/ru/darkkeks/vkmirror/bot/SetDescriptionAction.kt
package ru.darkkeks.vkmirror.bot

import ru.darkkeks.vkmirror.util.createLogger

// TODO Extract common class with SetNameAction
class SetDescriptionAction(autoReg: BotAutoReg,
                           val username: String,
                           val description: String) : BotFatherAction<Unit>(autoReg) {

    override suspend fun execute() {
        val newBot = autoReg.executeCommand(SET_DESCRIPTION)

        val timeout = getTimeout(newBot)
        if (timeout != null) {
            callback.completeExceptionally(BotFatherTimeoutException(timeout))
            return
        }

        val afterUsername = autoReg.executeCommand(username)

        if (INVALID_BOT_SELECTED.find(afterUsername) != null) {
            callback.completeExceptionally(IllegalStateException("Invalid username $username"))
        }

        autoReg.executeCommand(description)
        callback.complete(Unit)
    }

    companion object {
        val logger = createLogger()
    }
}